src/main/java/com/example/jpaboard/user/api/UserController.java

```diff
@@ -0,0 +1,43 @@
+package com.example.jpaboard.user.api;
+
+import com.example.jpaboard.user.domain.User;
+import com.example.jpaboard.common.ApiResponse;
+import com.example.jpaboard.user.dto.UserRequest;
+import com.example.jpaboard.user.dto.UserResponse;
+import com.example.jpaboard.exception.UserNotFoundException;
+import com.example.jpaboard.user.infra.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/users")
+@RequiredArgsConstructor
+public class UserController {
+
+    private final UserRepository userRepository;
+
+    @PostMapping
+    public ApiResponse<UserResponse> join(@RequestBody UserRequest userRequest) {
+        User user = new User(userRequest.getName(), userRequest.getHobby());
+        userRepository.save(user);
+        return ApiResponse.ok(new UserResponse(user.getId(), user.getName(), user.getHobby()));
+    }
```

Comment:
```suggestion
        User user = new User(userRequest.getName(), userRequest.getHobby());
        persistedUser = userRepository.save(user);
        return ApiResponse.ok(new UserResponse(persistedUser .getId(), persistedUser .getName(), persistedUser .getHobby()));
    }
```
영속된 객체를 사용하는게 흐름상 이해하기 더 좋네요.

또 이 부분은 컨트롤러가 무거워지는 것 같은데, 서비스레이어로 분리하면 좋을 것 같습니다.